### PR TITLE
C# 6.0 feature: Add library dependencies for C# 6.0

### PR DESCRIPTION
**Pull request by BillWagner**
New method to `System.String`
Additional types and methods

### CODE REVIEW

**Comment by BillWagner** [comment]
As part of reviewing this, we should cross-reference with the [feature tracking doc](https://github.com/dotnet/csharpstandard/blob/draft-v6/admin/v6-feature-tracker.md) to make these are the only new library dependencies.

**Comment by BillWagner** [comment]
> As part of reviewing this, we should cross-reference with the feature tracking doc to make these are the only new library dependencies.

@gafter Should this PR include an addition for `Array.Empty<T>()`?  See #200

**Comment by gafter** [comment]
Don't forget to add empty array API; see https://github.com/dotnet/csharpstandard/pull/5

**Comment by gafter** [comment]
> @gafter Should this PR include an addition for `Array.Empty<T>()`?  See #200 

I've added a note to #200 to add it at that time.

**Review by gafter — APPROVED** [review]

File: standard/standard-library.md [hunk 1]
```diff
@@ -628,4 +634,27 @@ namespace System.Runtime.CompilerServices
...
+    public class FormattableString : IFormattable
+    {
+    }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public static class FormattableStringFactory
+    {
+        public static FormattableString Create (string format, params object[] arguments);
```
Comment (jskeet) [inline standard/standard-library.md]: Please can we remove the space between `Create` and `(`?

**Review by jskeet — APPROVED** [review]

**Comment by jskeet** [comment]
(I have deja vu here... I thought I created basically this PR ages ago...)

**Comment by jskeet** [comment]
Bill - you can merge this after the nit-pick is fixed :)

**Comment by BillWagner** [comment]
> (I have deja vu here... I thought I created basically this PR ages ago...)

@jskeet You did the version for V5, IIRC.

File: standard/standard-library.md (hunk 1 above)
Comment (BillWagner) [reply standard/standard-library.md]:
```suggestion
        public static FormattableString Create(string format, params object[] arguments);
```

**Pull request merged**